app/src/main/java/de/westnordost/streetcomplete/controls/MainMenuButtonFragment.kt

```diff
@@ -38,12 +46,46 @@ class MainMenuButtonFragment : Fragment(R.layout.fragment_main_menu_button),
         super.onViewCreated(view, savedInstanceState)
 
         view.mainMenuButton.setOnClickListener { onClickMainMenu() }
+        teamModeColorCircle = view.teamModeColorCircle
+
+        if (teamModeQuestFilter.isEnabled) {
+            onTeamModeChanged(true)
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        teamModeQuestFilter.addListener(this)
+    }
+
+    override fun onStop() {
+        super.onStop()
+
+        teamModeQuestFilter.removeListener(this)
     }
 
     /* ------------------------------------------------------------------------------------------ */
 
     internal fun onClickMainMenu() {
-        context?.let { MainMenuDialog(it, this::onClickDownload).show() }
+        MainMenuDialog(
+            requireContext(),
+            if (teamModeQuestFilter.isEnabled) teamModeQuestFilter.indexInTeam else null,
+            this::onClickDownload,
+            teamModeQuestFilter::enableTeamMode,
+            teamModeQuestFilter::disableTeamMode
+        ).show()
+    }
+
+    override fun onTeamModeChanged(enabled: Boolean) {
+        if (enabled) {
+            requireContext().toast(R.string.team_mode_enabled, Toast.LENGTH_LONG)
+            teamModeColorCircle.popIn()
+            TeamModeColorCircle.setViewColorsAndText(teamModeColorCircle, teamModeQuestFilter.indexInTeam)
+        } else {
+            requireContext().toast(R.string.team_mode_disabled, Toast.LENGTH_SHORT)
+            teamModeColorCircle.popOut()
+        }
     }
```

Comment:
Should we invalidate the quests here when team mode gets disabled, as suggested in the original issue #2189?

If so, is `downloadedTilesDao.removeAll()` the right function to call? It doesn't seem start a new download immediately though.

Reply:
No, that's not the right function to call. That makes the app forget that it downloaded anything, so it downloads again everything on the next opportunity.

On invalidation, the `QuestPinLayerManager` which is a listener to the `VisibleQuestsSource` should simply call `clear()` on itself (=removes all the quests + forgets that it retrieved from database any quests). This should next time the map is moved re-load all the quests from db, i.e. make `QuestPinLayerManager` to ask `VisibleQuestsSource` for quests.
Maybe to make this happen immediately and not just after the user panned the map, call `onNewScreenPosition()` after `clear()`.

Reply:
> That makes the app forget that it downloaded anything, so it downloads again everything on the next opportunity.

That's the idea though. From #2189:

> After exiting the team modus again, those areas' quest cache should be invalidated automatically to make sure that the quests solved by the other team members won't be displayed the next time the user want to map there.

So we actually want to re-download the quests to make sure that other team members' solved quests don't show up after exiting team mode.

The changes in `QuestPinLayerManager` are already implemented in [`68d3a2c` (#2557)](https://github.com/streetcomplete/StreetComplete/pull/2557/commits/68d3a2cce0bf82ae2fc8a63371d5221acb595279).

Reply:
> That's the idea though. From #2189:

>>    After exiting the team modus again, those areas' quest cache should be invalidated automatically to make sure that the quests solved by the other team members won't be displayed the next time the user want to map there.

Ah right, makes sense. Hmm, but it only "invalidates" them, so they won't actually get updated after the next download. The `QuestAutoSyncer ` does the automatic downloading. So, that class should maybe also listen to changes on the team mode state. If the team mode got turned off, it should trigger an auto-download (after `downloadedTilesDao.removeAll()` has been called).